### ISSUE TITLE
Update to new http-parser API

### ISSUE THREAD

**Issue by jonashaag** [jonashaag/bjoern#50]
They removed the path and query callbacks and added a URL parsing utility instead.

**Comment by sideffect0** [comment]
@jonashaag what is it's status now ?

**Comment by jonashaag** [comment]
I'm happy to see this done by someone else :-) There's no pressure for this, though, since the old version we're using works really well.

**Comment by sideffect0** [comment]
how tests are organised, & what about development dependencies ?

**Comment by jonashaag** [comment]
There aren't many real unittests -- most testing is done manually by simply starting some WSGI application and using a real browser.

> what about development dependencies ?

What do you mean by that?

**Comment by jelly** [comment]
I would really like to package bjoern for Debian/Arch and the bundling of http-parser is generally "not done" for these distro's. I've made a branch with one commit c93574eb19e0614787fca0c7ea7bb07e8d1da687 to make it compatible with http-parser by renaming the offending enum, this leaves two minor compiler warnings.

My second more "controversial" change removes the bundling, making building it with system libs optional would be fine for me too. fe4621635119d616e4a488804fe912de31bef245

**Comment by jelly** [comment]
Update, I tested it further and it segfaults now with our full API testsuite:
```
segfault at a ip 00007fe76061da12 sp 00007ffca4e2a3f0 error 4 in _bjoern.cpython-37m-x86_64-linux-gnu.so[7fe76061d000+3000]```
I'll have to do more digging

**Comment by jelly** [comment]
@jonashaag any comments? tips?

**Comment by jonashaag** [comment]
Sorry what’s your question exactly? There’s some kind of migration tips in the http parser documentation (not sure where but you’ll find it ;)

**Comment by jelly** [comment]
Ah, well I couldn't find any migration documentation on https://github.com/nodejs/http-parser and there is a bugreport open for a CHANGES file.

**Comment by jonashaag** [comment]
You're right, it's kind of difficult to find. There's some info here: https://github.com/nodejs/http-parser/pull/54 and some here: https://github.com/nodejs/http-parser/pull/58

I guess the best way is to compare the old README with the current one...
